src/containers/BooksList/BooksList.js

```diff
@@ -0,0 +1,59 @@
+import React, {Component} from 'react'
+import {connect} from "react-redux";
+import PropTypes from "prop-types";
+
+import { BOOK_CATEGORIES } from "../../logic/book";
+import CategoryFilter from "../../components/CategoryFilter/CategoryFilter";
+import { changeFilter } from "../../actions";
+import {removeBook} from "../../actions";
+import Book from "../../components/Book/Book";
+
+class BooksList extends Component {
+    static propTypes = {
+        dispatchRemoveBook: PropTypes.func.isRequired,
+        dispatchChangeFilter: PropTypes.func.isRequired,
+    }
+    _removeBook = (book) => {
+        this.props.dispatchRemoveBook(book)
+    }
+    _getBooksToRender = () => {
+        const {books} = this.props.bookstore
+        const filterCategoryIndex = this.props.filter.filter;
+
+        return books.filter((book, index) => {
+            if (filterCategoryIndex === -1) {
+                return true
+            }
+            if (filterCategoryIndex === BOOK_CATEGORIES.indexOf(book.category)) {
+                return true
+            }
+            return false
+        });
+    }
+    _handleFilterChange = (category) => {
+        this.props.dispatchChangeFilter(category);
+    }
+
+    render() {
+        return (
+            <div>
+                <CategoryFilter handleFilterChange={this._handleFilterChange} />
+                {this._getBooksToRender().map((book, index) => <Book book={book} key={index} removeBook={this._removeBook}/> )}
```

Comment:
This actually is what is making it messy. The `_getBooksToRender` function isn't necessary to handle this.

Why didn't we just use `mapStateToProps` to filter out the books we wanted to render? Like so, for example:

```
const mapStateToProps => state => ({
  filter: state.filter,
  books: state.booksStore.books.filter(book => <filter out books based on value of state.filter>)
})
```
Then you won't need those unnecessary extra functions just to get this done. Your code would be much cleaner